setup/src/Magento/Setup/Controller/Navigation.php

```diff
@@ -32,14 +33,20 @@ class Navigation extends AbstractActionController
      */
     protected $view;
 
+    /**
+     * @var ObjectManagerProvider $objectManagerProvider
+     */
+    protected $objectManagerProvider;
+
     /**
      * @param NavModel $navigation
      * @param Status $status
      */
-    public function __construct(NavModel $navigation, Status $status)
+    public function __construct(NavModel $navigation, Status $status, ObjectManagerProvider $objectManagerProvider)
```

Comment:
Why do you use `ObjectManagerProvider` instead of injecting just `UrlInterface`?

Reply:
I am getting such error "An error occurred during execution; please try again later" and status 500 while injecting UrlInterface directly in constructor of controller